content/en/docs/how-tos/cluster-claim.md

```diff
@@ -0,0 +1,62 @@
+---
+title: "Creating a Cluster Pool"
+description: How to create a cluster pool from which a job can claim a cluster.
+---
+
+DPTP integrates Hive into the OpenShift test platform to manage the cluster pools and clusters in them. There is a defined pool [ci-cluster-pool](https://github.com/openshift/release/tree/master/clusters/app.ci/ci-cluster-pool) from which can be used to claim an OpenShift cluster for a test. In this page, we document how to create a new cluster pool with the cloud account provided by a CI user.
+
+## Prerequisites
+
+*  Choose a [Hive](https://github.com/openshift/hive#supported-cloud-providers) supported cloud for the pool.
+* [Configure the cloud account](https://docs.openshift.com/container-platform/4.7/installing/installing-preparing.html).
+
+## Create a cluster pool
+
+Create a folder in `release/clusters/app.ci/` with the manifests for the pool. See [the examples](https://github.com/openshift/release/tree/master/clusters/app.ci/ci-cluster-pool) for the existing `ci-cluster-pool`.
+
+```yaml
+apiVersion: hive.openshift.io/v1
+kind: ClusterPool
+metadata:
+  name: ci-ocp-4-7-amd64-aws-us-east-1
+  namespace: cvp-cluster-pool
+  labels: # product, version, architecture, cloud, owner are used to filter out a pool when a job claims a cluster
+    product: ocp
+    version: "4.7"
+    architecture: amd64
+    cloud: aws
+    owner: cvp
+    region: us-east-1
+spec:
+  baseDomain: hive.example.org # the base domain to install the cluster
+  imageSetRef:
+    name: ocp-4.7.0-amd64 # the name of the imageSet which determines the image to install the cluster
+  installConfigSecretTemplateRef:
+    name: install-config-aws-us-east-1 # the name of the secret with an installation config for the installer
+  skipMachinePools: true
+  platform:
+    aws:
+      credentialsSecretRef:
+        name: hive-aws-credentials # the name of the secret with the credentials of the cloud account
+      region: us-east-1
+  pullSecretRef:
+    name: pull-secret # the name of the pull secret used by installer
```

Comment:
Is this something that a user needs to set? AFAIK we need everyone to use our pull-secrets (so they can pull ci-op built images,etc), no?

Reply:
No.
We only install GA version of OpenShift via pools. No CI images are needed.

The pull secret can be obtained as described https://docs.openshift.com/container-platform/4.7/installing/installing_aws/installing-aws-default.html

Reply:
> We only install GA version of OpenShift via pools. No CI images are needed. 

That's not true. If a user wants to test their operator, they will build it in CI and the test cluster must be able to install it.

Reply:
Interesting. In principle, this isnt the concern of the pool.
This is day 2 operation (after installation), I guess.
The test has admin access to the cluster. They can modify the global pull secret if needed.

For example, my test cases (I know very unlikely) are
Case1: my operator pod does not run without linking the pull secret to the service account running the pod.

Case2: my operator pod runs after linking the pull secret to the service account running the pod.

I just realize that this become tricky.
Where are the images built in CI for an operator used?
- image of a pod
- input of a build
- something else

How are they referred?
- build01.registry.ci.openshft.org/ci-op-xxxx/component:tag
- image-registry.openshift-image-registry.svc:5000/ci-op-xxxx/component:tag

In the case of latter, it might be hard to inject it in advance because it is not clear from which cluster the token should be taken.

Reply:
I saw many templates make the images in ci-op-xxxx public.
We could do the same in for oo tests.

Reply:
How come this works?
https://github.com/openshift/release/pull/17798/files
Does it install any image built in CI?

Reply:
Huh, right. IIRC there are steps that expose the images, indeed. Perfect, let's keep it this way. At worst we iterate.